libs/agno/tests/integration/workflow/test_streaming_events.py

```diff
@@ -0,0 +1,851 @@
+"""Integration tests for workflow streaming events."""
+
+from typing import List
+
+import pytest
+
+from agno.run.v2.workflow import (
+    ConditionExecutionCompletedEvent,
+    ConditionExecutionStartedEvent,
+    LoopExecutionCompletedEvent,
+    LoopExecutionStartedEvent,
+    LoopIterationCompletedEvent,
+    LoopIterationStartedEvent,
+    ParallelExecutionCompletedEvent,
+    ParallelExecutionStartedEvent,
+    RouterExecutionCompletedEvent,
+    RouterExecutionStartedEvent,
+    StepCompletedEvent,
+    StepsExecutionCompletedEvent,
+    StepsExecutionStartedEvent,
+    StepStartedEvent,
+    WorkflowCompletedEvent,
+    WorkflowStartedEvent,
+)
+from agno.utils.log import logger
+from agno.workflow.v2 import Condition, Loop, Parallel, Router, Step, Steps, Workflow
+from agno.workflow.v2.types import StepInput, StepOutput
+
+
+# Helper functions for testing
+def step_function(step_input: StepInput) -> StepOutput:
+    """Basic step function."""
+    return StepOutput(content=f"Step output: {step_input.message}")
+
+
+def simple_step_1(step_input: StepInput) -> StepOutput:
+    return StepOutput(content="Step 1 completed")
+
+
+def simple_step_2(step_input: StepInput) -> StepOutput:
+    return StepOutput(content="Step 2 completed")
+
+
+def condition_check(step_input: StepInput) -> bool:
+    """Simple condition check."""
+    return "test" in str(step_input.message)
+
+
+def loop_condition(outputs):
+    """Loop end condition."""
+    return len(outputs) >= 2
+
+
+def router_function(step_input: StepInput) -> List[Step]:
+    """Router function to select steps."""
+    if "a" in str(step_input.message):
+        return [Step(name="route_a", executor=step_a)]
+    else:
+        return [Step(name="route_b", executor=step_b)]
+
+
+def simple_router(step_input: StepInput) -> list:
+    return [Step(name="routed_step", executor=simple_step_1)]
+
+
+def step_a(step_input: StepInput) -> StepOutput:
+    """Step A for router."""
+    return StepOutput(content="Step A executed")
+
+
+def step_b(step_input: StepInput) -> StepOutput:
+    """Step B for router."""
+    return StepOutput(content="Step B executed")
+
+
+# ============================================================================
+# STEP EVENTS TESTS
+# ============================================================================
+
+def test_step_events_with_stream_intermediate_steps_false(workflow_storage):
+    """Test Step events with stream_intermediate_steps=False."""
+    workflow = Workflow(
+        name="Step Events Test",
+        storage=workflow_storage,
+        steps=[Step(name="test_step", executor=step_function)],
+    )
+
+    events = list(workflow.run(message="test", stream=True, stream_intermediate_steps=False))
+    event_types = [type(event).__name__ for event in events]
+
+    # Should only have workflow events, NO step started/completed events
+    assert "WorkflowStartedEvent" in event_types
+    assert "WorkflowCompletedEvent" in event_types
```

Comment:
Typically we wouldn't have these.... but it would break if you have NO events. So I guess this is safe.